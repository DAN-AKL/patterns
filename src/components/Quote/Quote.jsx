import React from "react";
import * as PropTypes from "prop-types";

import "./Quote.scss";

const propTypes = {
  text: PropTypes.string,
  name: PropTypes.string,
  role: PropTypes.string
};

class Quote extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <blockquote className="quote">
        <p className="quote__text">{this.props.text}</p>
        <cite className="quote__citation">
          <span className="quote__name">{this.props.name}</span>
          <span className="quote__role">{this.props.role}</span>
        </cite>
      </blockquote>
    );
  }
}

Quote.propTypes = propTypes;

export default Quote;
