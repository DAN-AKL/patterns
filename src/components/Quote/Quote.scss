.quote {
}
.quote__text {
  font-size: 15rem;
  font-weight: 800;
  line-height: 1.4;

  &:before,
  &:after {
    content: '"';
  }
}

.quote__citation {
  border-left: 0.5rem solid #000;
  display: block;
  font-size: 5rem;
  margin-top: 4rem;
  padding-left: 4rem;
}

.quote__name,
.quote__role {
  display: block;
}
