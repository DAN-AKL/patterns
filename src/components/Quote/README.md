## Header

This is some information about the Quote
