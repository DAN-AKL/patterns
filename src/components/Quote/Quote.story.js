import React from "react";

import Quote from "./Quote.jsx";
import { storiesOf } from "@storybook/react";
import { withKnobs, text } from "@storybook/addon-knobs";

const stories = storiesOf("Quote", module);

stories.addDecorator(withKnobs);

stories.add("Default", () => {
  return (
    <Quote
      text="DAN was awesome during our project, I thoroughly enjoyed working with them. Everyone should use them."
      name="Sarah Lastname "
      role="Auckland Company Ltd"
    />
  );
});
