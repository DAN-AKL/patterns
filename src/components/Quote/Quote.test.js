import React from "react";
import renderer from "react-test-renderer";
import Quote from "./Quote.jsx";

it("renders a quote component", () => {
  const tree = renderer
    .create(
      <Quote
        text="DAN was awesome during our project, I thoroughly enjoyed working with them. Everyone should use them."
        name="Sarah Lastname"
        role="Auckland Company Ltd"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
