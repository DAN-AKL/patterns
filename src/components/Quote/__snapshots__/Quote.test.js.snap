// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a quote component 1`] = `
<blockquote
  className="quote"
>
  <p
    className="quote__text"
  >
    DAN was awesome during our project, I thoroughly enjoyed working with them. Everyone should use them.
  </p>
  <cite
    className="quote__citation"
  >
    <span
      className="quote__name"
    >
      Sarah Lastname
    </span>
    <span
      className="quote__role"
    >
      Auckland Company Ltd
    </span>
  </cite>
</blockquote>
`;
